src/rust/receiver-mock/src/logs.rs

```diff
@@ -1,74 +1,198 @@
-use std::collections::HashMap;
+use anyhow::anyhow;
+use serde_json::Value;
+use std::collections::{BTreeMap, HashMap};
 use std::net::IpAddr;
 
-#[derive(Clone)]
+#[derive(Clone, Debug, PartialEq)]
 pub struct LogStats {
-    pub count: u64,
-    pub bytes: u64,
+    pub message_count: u64,
+    pub byte_count: u64,
 }
 
-#[derive(Clone)]
-pub struct LogRepository {
+pub struct LogStatsRepository {
     pub total: LogStats,
-    pub ipaddr_to_stats: HashMap<IpAddr, LogStats>,
+    pub ipaddr: HashMap<IpAddr, LogStats>,
 }
 
-impl LogRepository {
+impl LogStatsRepository {
     pub fn new() -> Self {
         return Self {
-            total: LogStats { count: 0, bytes: 0 },
-            ipaddr_to_stats: HashMap::new(),
+            total: LogStats {
+                message_count: 0,
+                byte_count: 0,
+            },
+            ipaddr: HashMap::new(),
         };
     }
 
-    pub fn add_log_message(&mut self, body: String, ipaddr: IpAddr) {
-        self.total.count += 1;
-        self.total.bytes += body.len() as u64;
-        let stats = self
-            .ipaddr_to_stats
-            .entry(ipaddr)
-            .or_insert(LogStats { count: 0, bytes: 0 });
-        stats.count += 1;
-        stats.bytes += body.len() as u64;
+    pub fn update(&mut self, message_count: u64, byte_count: u64, ipaddr: IpAddr) {
+        // update total stats
+        self.total.message_count += message_count;
+        self.total.byte_count += byte_count;
+
+        // update per ip address stats
+        let stats = self.ipaddr.entry(ipaddr).or_insert(LogStats {
+            message_count: 0,
+            byte_count: 0,
+        });
+        stats.message_count += message_count;
+        stats.byte_count += byte_count as u64;
     }
 
+    #[cfg(test)]
     pub fn get_stats_for_ipaddr(&self, ipaddr: IpAddr) -> LogStats {
         return self
-            .ipaddr_to_stats
+            .ipaddr
             .get(&ipaddr)
-            .unwrap_or(&LogStats { count: 0, bytes: 0 })
+            .unwrap_or(&LogStats {
+                message_count: 0,
+                byte_count: 0,
+            })
             .clone();
     }
 }
 
+#[derive(Clone)]
+pub struct LogMessage {
+    // This structure is intended to house more data as we add APIs requiring it
+// For example, metadata when we want to query log count by label
```

Comment:
`rustfmt` insists on formatting this comment this way, not sure why. I'm going to remove it soon, so it shouldn't matter much.